### PR TITLE
xtensa/esp32: Add WPA2 Enterprise and WPA3 support

### PR DESCRIPTION
**Pull request by donghengqaz**
## Summary

Add WPA2 Enterprise and WPA3 support.

### CODE REVIEW

File: arch/xtensa/src/esp32/esp32_wifi_adapter.c [hunk 1]
```diff
@@ -4201,8 +4199,6 @@ int32_t esp_timer_start_periodic(esp_timer_handle_t timer, uint64_t period)
 {
   struct rt_timer_s *rt_timer = (struct rt_timer_s *)timer;
 
-  DEBUGASSERT(period <= UINT32_MAX);
```
Comment (acassis) [inline arch/xtensa/src/esp32/esp32_wifi_adapter.c:4204]: Shouldn't be replaced with UINT64_MAX ?

Comment (yamt) [reply arch/xtensa/src/esp32/esp32_wifi_adapter.c:4204]: @acassis isn't it always true for a uint64_t value?

**Comment by yamt** [comment]
how is timeout uint64_t change related to "WPA2 Enterprise and WPA3"? just a curious question.

File: arch/xtensa/src/esp32/esp32_wifi_adapter.c (hunk 1 above)
Comment (donghengqaz) [reply arch/xtensa/src/esp32/esp32_wifi_adapter.c:4204]: period must be <= UINT64_MAX.

**Comment by donghengqaz** [comment]
> how is timeout uint64_t change related to "WPA2 Enterprise and WPA3"? just a curious question.

Because the WPA supplicant for WPA3 will create a timer and its timeout time is after 100 days = 0x7DB A821 8000 us.

I have adviced my company's WiFi team to think out using OS timer not real-time timer, but it needs some time.

File: arch/xtensa/src/esp32/esp32_wifi_adapter.c (hunk 1 above)
Comment (acassis) [reply arch/xtensa/src/esp32/esp32_wifi_adapter.c:4204]: Yes, you are right, I missed that period is a uint64_t.

**Pull request merged**